strutil/strutil.go

```diff
@@ -111,3 +111,37 @@ func TruncateOutput(data []byte, maxLines, maxBytes int) []byte {
 	}
 	return data
 }
+
+// ParseValueWithUnit parses a value like 500kB and returns the number
+// in byte
+func ParseValueWithUnit(inp string) (int64, error) {
+	unitMultiplier := map[string]int{
+		"B":  1,
+		"kB": 1000,
+		"MB": 1000 * 1000,
+		"GB": 1000 * 1000 * 1000,
+		"TB": 1000 * 1000 * 1000 * 1000,
+		"PB": 1000 * 1000 * 1000 * 1000 * 1000,
+		"EB": 1000 * 1000 * 1000 * 1000 * 1000 * 1000,
+	}
+	if len(inp) < 2 {
+		return 0, fmt.Errorf("cannot parse %q: need a unit", inp)
+	}
+	// simple case, "1234B", read as byte
+	if lastCh := inp[len(inp)-1]; lastCh == 'B' {
```

Comment:
Unless I'm missing something, we will always hit this case since every unit ends with 'B', so we will (usually) unnecessarily try convert the string to int, it will fail so we will carry on with two-char suffix check.
Since two-letter units will be used most of the time I suppose, would it make sense to reverse and re-organise the two conditions (`unitMultiplier[unit]` lookup failure wouldn't be fatal in such case), or alternatively enhance this one to check if the inp[len(inp)-2] character is a digit to avoid ParseInt?

Reply:
Yes but then we don't do anything when err != nil (we just continue and then hit the remaining cases)

Reply:
Yes, what I'm trying to say is we will hit ParseInt error case 99% of the time if we check for 'B' suffix first.

Reply:
And to be clear, the logic as currently is is perfectly correct, just suggesting to slightly re-organize the checks so that we only attempt ParseInt when we know unit is already correct (after two-character lookup, and then 'B' suffix check).

Reply:
I changed the order of the checks now.